### PR TITLE
TEST/UCM/JENKINS: Extend memtype hook tests

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: test/apps/test_cuda_hook.c [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+{
+    int *count_p = arg;
+    const char *title;
+
+    if (event_type == UCM_EVENT_MEM_TYPE_ALLOC) {
+        title = "allocate";
+    } else if (event_type == UCM_EVENT_MEM_TYPE_FREE) {
+        title = "free";
+    } else {
+        /* don't care */
```
Comment (hoopoepg) [inline test/apps/test_cuda_hook.c:26]: minor: may be ```printf ("unknown type %d", event_type);```?

File: test/apps/test_cuda_hook.c [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    status            = ucp_init(&params, NULL, &context);
+    if (status != UCS_OK) {
+        printf("failed to create context\n");
+        return -1;
+    }
+
+    num_events = 0;
+    ucm_set_event_handler(memtype_events, 1000, event_cb, &num_events);
+
+    {
```
Comment (hoopoepg) [inline test/apps/test_cuda_hook.c:57]: C++ pattern to define local class?

**Review by hoopoepg — APPROVED** [review]
minor comments

**Comment by yosefe** [comment]
@hoopoepg fixed

**Review by hoopoepg — APPROVED** [review]

**Pull request merged**